pandas/tests/test_algos.py

```diff
@@ -2409,3 +2409,10 @@ def test_diff_ea_axis(self):
         msg = "cannot diff DatetimeArray on axis=1"
         with pytest.raises(ValueError, match=msg):
             algos.diff(dta, 1, axis=1)
+
+    @pytest.mark.parametrize("dtype", ["int8", "int16"])
```

Comment:
can you also add a replica of the test from the OP in groupby tests. I don't think we actually have any :sigh:

pandas/tests/groupby//test_groupby.py near shift is fine (or actually ok with moving the shift & diff tests out to a new file).

pls add for all dtypes

Reply:
Added test from issue and for other dtypes